com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ValidateDeployParametersStep.java

```diff
@@ -48,32 +58,72 @@ protected String getStepErrorMessage(DelegateExecution context) {
     }
 
     private void validateParameters(DelegateExecution context) {
-        validateStartTimeout(context);
-        validateAppArchiveId(context);
-        validateExtDescriptorFileId(context);
-        validateVersionRule(context);
+        try {
+            validateStartTimeout(context);
+            validateAppArchiveId(context);
+            validateExtDescriptorFileId(context);
+            validateVersionRule(context);
+            verifyArchive(context);
+        } finally {
+            deleteArchiveFile(context);
+        }
     }
 
-    private void validateVersionRule(DelegateExecution context) {
-        String versionRuleString = (String) context.getVariable(Constants.PARAM_VERSION_RULE);
+    private void validateStartTimeout(DelegateExecution context) {
+        Object parameter = context.getVariable(Constants.PARAM_START_TIMEOUT);
+        if (parameter == null) {
+            return;
+        }
+        int startTimeout = (Integer) parameter;
+        if (startTimeout < 0) {
+            throw new SLException(Messages.ERROR_PARAMETER_1_MUST_NOT_BE_NEGATIVE, startTimeout, Constants.PARAM_START_TIMEOUT);
+        }
+    }
+
+    private void validateAppArchiveId(DelegateExecution context) {
+        String appArchiveId = StepsUtil.getRequiredString(context, Constants.PARAM_APP_ARCHIVE_ID);
+        String[] appArchivePartsId = appArchiveId.split(",");
+        List<FileEntry> archivePartEntries = getArchivePartEntries(context, appArchivePartsId);
+        StepsUtil.setAsJsonBinaries(context, Constants.VAR_FILE_ENTRIES, archivePartEntries);
+        getStepLogger().debug(Messages.BUILDING_ARCHIVE_FROM_PARTS);
+        new ExecutionRetrier().executeWithRetry(createArchiveFromParts(context, archivePartEntries));
+    }
+
+    private List<FileEntry> getArchivePartEntries(DelegateExecution context, String[] appArchivePartsId) {
+        List<FileEntry> archivePartEntries = new ArrayList<>();
+        for (String appArchivePartId : appArchivePartsId) {
+            archivePartEntries.add(findFile(context, appArchivePartId));
+        }
+        return archivePartEntries;
```

Comment:
This could be substituted with Arrays.stream().map().collect()

Reply:
Done. Substituted.